cmd/lakectl/cmd/bisect.go

```diff
@@ -0,0 +1,375 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/pkg/api"
+)
+
+const bisectCommitTemplate = `{{ range $val := . }}
+ID:            {{ $val.Id|yellow }}{{if $val.Committer }}
+Author:        {{ $val.Committer }}{{end}}
+Date:          {{ $val.CreationDate|date }}
+Meta Range ID: {{ $val.MetaRangeId }}
+{{ if gt ($val.Parents|len) 1 -}}
+Merge:         {{ $val.Parents|join ", "|bold }}
+{{ end }}
+	{{ $val.Message }}
+{{ if $val.Metadata.AdditionalProperties }}
+Metadata:
+	{{ range $key, $value := $val.Metadata.AdditionalProperties }}
+	{{ $key | printf "%-18s" }} = {{ $value }}
+	{{ end -}}
+{{ end -}}
+{{ end }}`
+
+type BisectSelect int
+
+const (
+	BisectSelectGood BisectSelect = iota
+	BisectSelectBad
+)
+
+var ErrCommitNotFound = errors.New("commit not found")
+
+// bisectCmd represents the bisect command
+var bisectCmd = &cobra.Command{
+	Use:   "bisect",
+	Short: "Binary search to find the commit that introduced a bug",
+}
+
+type Bisect struct {
+	Created    time.Time     `json:"created"`
+	Repository string        `json:"repository"`
+	BadCommit  string        `json:"badCommit,omitempty"`
+	GoodCommit string        `json:"goodCommit,omitempty"`
+	Commits    []*api.Commit `json:"commits,omitempty"`
+}
+
+const bisectStartCmdArgs = 3
+
+// bisectStartCmd represents the start command
+var bisectStartCmd = &cobra.Command{
+	Use:   "start <repository> <bad ref> <good ref>",
```

Comment:
Why do we need the repo if we have the refs?

Reply:
At first the two refs were optional - fixing.